openfga_sdk/client/client.py

```diff
@@ -629,12 +643,124 @@ async def batch_check(
+            api_response = await self._api.batch_check(body, **kwargs)
+            return api_response
+        except Exception as err:
+            raise err
+        finally:
+            semaphore.release()
+
+    async def batch_check(self, body: ClientBatchCheckRequest, options=None):
+        """
+        Run a batchcheck request
+        :param body - BatchCheck request
+        :param authorization_model_id(options) - Overrides the authorization model id in the configuration
+        :param max_parallel_requests(options) - Max number of requests to issue in parallel. Defaults to 10
+        :param max_batch_size(options) - Max number of checks to include in a request. Defaults to 50
+        :param header(options) - Custom headers to send alongside the request
+        :param retryParams(options) - Override the retry parameters for this request
+        :param retryParams.maxRetry(options) - Override the max number of retries on each API request
+        :param retryParams.minWaitInMs(options) - Override the minimum wait before a retry is initiated
+        """
+        options = set_heading_if_not_set(
+            options, CLIENT_BULK_REQUEST_ID_HEADER, str(uuid.uuid4())
+        )
+
+        max_parallel_requests = 10
+        if options is not None and "max_parallel_requests" in options:
+            if (
+                isinstance(options["max_parallel_requests"], str)
+                and options["max_parallel_requests"].isdigit()
+            ):
+                max_parallel_requests = int(options["max_parallel_requests"])
+            elif isinstance(options["max_parallel_requests"], int):
+                max_parallel_requests = options["max_parallel_requests"]
+
+        max_batch_size = 50
+        if options is not None and "max_batch_size" in options:
+            if (
+                isinstance(options["max_batch_size"], str)
+                and options["max_batch_size"].isdigit()
+            ):
+                max_batch_size = int(options["max_batch_size"])
+            elif isinstance(options["max_batch_size"], int):
+                max_batch_size = options["max_batch_size"]
+
+        check_to_id: dict[str, ClientBatchCheckItem] = {}
+
+        def track_and_transform(checks):
+            transformed = []
+            for check in checks:
+                if check.correlation_id is None:
+                    check.correlation_id = str(uuid.uuid4())
+
+                if check.correlation_id in check_to_id:
+                    raise FgaValidationException("Duplicate correlation_id provided")
+
+                check_to_id[check.correlation_id] = check
+
+                transformed.append(construct_batch_item(check))
+            return transformed
+
+        checks = [
+            track_and_transform(
+                body.checks[i * max_batch_size : (i + 1) * max_batch_size]
+            )
+            for i in range((len(body.checks) + max_batch_size - 1) // max_batch_size)
+        ]
```

Comment:
do we raise an error if the request contains empty checks?

Reply:
Ah, my assumption was that this would be handled by the API, but as we don't make any calls if we aren't passed checks then that won't happen.

Maybe this is one of those where we _do_ have some client side validation and error if no checks are passed?

Reply:
Yeah for both async/sync, `write` will just do nothing and return a `ClientWriteResponse` with empty `writes` and `deletes` lists. With `batch_check` for both async/sync we will return a `ClientBatchCheckResponse` with an empty `checks` list.

Reply:
>  With batch_check for both async/sync we will return a ClientBatchCheckResponse with an empty checks list.

Are we sure about that? When I call the API with either an empty checks list, or no checks field at all, I get a 400 validation error. I assume in the absence of client-side validation (which likely is consistent with other methods), the same error would be returned.

Reply:
Ah, I think I see what you are saying, that the SDK won't actually make a request if there are no checks, and just return an empty ClientBatchCheckResponse?

Reply:
Yup exactly that, as we're batching the BatchCheck calls, we end up with no calls to the API and a `ClientBatchCheckResponse` with empty `checks`